lua/coverage/init.lua

```diff
@@ -63,7 +63,14 @@ M.load = function(place)
     if lang_config == nil then
         lang_config = config.opts.lang[lang.config_alias]
     end
-    if lang_config ~= nil and lang_config.coverage_file ~= nil then
+
+    -- Automatically watch the coverage file for updates when auto_reload is enabled
+    -- and when the language setup allows it:
+    --  - julia is disabled because the coverage command itself produces the file to be
+    --    watched which leads to an infinite loop (see
+    --    https://github.com/andythigpen/nvim-coverage/issues/41)
+    if config.opts.auto_reload and ftype ~= "julia" and
```

Comment:
good catch. i'm not sure why `auto_reload` wasn't checked here already.
i am going to merge this PR but i think i'll move the file type check into the config in a subsequent commit.